pkg/content/ported.go

```diff
@@ -218,7 +224,115 @@ func push(ctx context.Context, store content.Store, pushCtx *containerd.RemoteCo
 		limiter = semaphore.NewWeighted(int64(pushCtx.MaxConcurrentUploadedLayers))
 	}
 
-	return remotes.PushContent(ctx, pusher, desc, store, limiter, pushCtx.PlatformMatcher, wrapper)
+	return pushContent(ctx, pusher, desc, store, limiter, pushCtx.PlatformMatcher, wrapper, content)
+}
+
+// Ported from containerd project, copyright The containerd Authors.
+// https://github.com/containerd/containerd/blob/main/remotes/handlers.go
+func pushContent(ctx context.Context, pusher remotes.Pusher, desc ocispec.Descriptor, store content.Provider, limiter *semaphore.Weighted, platform platforms.MatchComparer, wrapper func(h images.Handler) images.Handler, localContent *Content) error {
+	var m sync.Mutex
+	manifests := []ocispec.Descriptor{}
+	indexStack := []ocispec.Descriptor{}
+
+	filterHandler := images.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+		switch desc.MediaType {
+		case images.MediaTypeDockerSchema2Manifest, ocispec.MediaTypeImageManifest:
+			m.Lock()
+			manifests = append(manifests, desc)
+			m.Unlock()
+			return nil, images.ErrStopHandler
+		case images.MediaTypeDockerSchema2ManifestList, ocispec.MediaTypeImageIndex:
+			m.Lock()
+			indexStack = append(indexStack, desc)
+			m.Unlock()
+			return nil, images.ErrStopHandler
+		default:
+			return nil, nil
+		}
+	})
+
+	pushHandler := remotes.PushHandler(pusher, store)
+	pushHandler = images.Handlers(pushHandler, images.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+		return nil, localContent.UpdateTime(&desc.Digest)
```

Comment:
The updatedat timestamp should be updated once the blob is written to the local disk, instead of pushing stage, right?

Reply:
How to know when the converted blob is written? (estargz/nydus)Convert has any hooks?

Reply:
Maybe we can override the method which needs to be hooked like containerd metadata [DB](https://github.com/containerd/containerd/blob/f3a07934b49bf142925a5913e3e19f3528eda0d2/metadata/content.go#L62).